### ISSUE TITLE
Resizing doesn't work under wayland

### ISSUE THREAD

**Issue by justDeeevin** [iced-rs/iced#2256]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

When I call `window::resize()`, the window fails to resize. It changes for what appears to be a frame, and then reverts back to its original state.

### What is the expected behavior?

The window with the specified ID should shape itself to the given size.

### Version

crates.io release

### Operating System

Linux

**Comment by JL710** [comment]
Can you reproduce that with the latest master? On my system it is not reproducible.

**Comment by piuvas** [comment]
not working here on:
niri 26.04
iced 0.14.0

**Comment by m3x3** [comment]
I'm having the same issue on Pop!_OS 24.04 with COSMIC DE (wayland) and iced from master (and 0.14). Did some digging and found that winit's [request_inner_size](https://docs.rs/winit/latest/winit/window/struct.Window.html#method.request_inner_size) returns `Some(<old size>)`, which would indicate that the platform doesn't allow resizing. The same code works without issues on Mint 21.2 (x11).

So not really iced's fault, at least in my case, but a platform limitation somewhere between winit, COSMIC, and wayland. Some kind of feedback that resizing failed could be helpful though.

**Comment by Ph4ntomas** [comment]
FWIW, resizing a window on linux has always been up to the window manager (for x11) or compositor (wayland). Usually tiling window manager will forcefully resize your window, or clip it if the client isn't well behaved.

Niri and cosmic are tiling window manager, so the behavior reported here is kinda expected, but even for floating window, the compositor are allowed to refuse clients requests.

**Comment by m3x3** [comment]
> Niri and cosmic are tiling window manager, so the behavior reported here is kinda expected

I agree to some degree; however AIUI COSMIC supports both tiling and floating layout, my workspace is in "floating" mode, and I can freely resize the window with the mouse, so I did find the limitation somewhat surprising. But in any case that's not an iced issue.

**Comment by Ph4ntomas** [comment]
> my workspace is in "floating" mode, and I can freely resize the window with the mouse, so I did find the limitation somewhat surprising. But in any case that's not an iced issue.

User-driven window resizing is a bit different, as the client request the resize to start as an answer to a user action, then the compositor may accept that request before sending the new size back to the client, so the final size is still decided by the compositor.

WRT floating mode, I don't know if cosmic implemented a different policy for when the window is in floating mode, but if you're primarily designing with tiling in mind and add floating on top, it's easier to just "ignore" client requests by default, especially since it's explicitely allowed by the protocol.

But either way, I'm not even sure this is technically a cosmic issue either, as it's perfectly within spec. Though definitely unusual if you're used to other platforms
